src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -194,7 +194,17 @@ private SecurityStatusPal EncryptData(ReadOnlyMemory<byte> buffer, ref byte[] ou
 
             if (NetSecurityTelemetry.Log.IsEnabled())
             {
-                return ProcessAuthenticationWithTelemetry(isAsync, isApm, cancellationToken);
+                ValueTask task = ProcessAuthenticationWithTelemetryAsync(isAsync, isApm, cancellationToken);
+                if (isAsync)
+                {
+                    return task.AsTask();
+                }
+                else
+                {
+                    Debug.Assert(task.IsCompleted);
+                    task.GetAwaiter().GetResult();
+                    return null;
```

Comment:
I don't remember why we did this, but seeing it again now, it'd be cleaner/simpler to just always return the Task and then have the synchronous call sites do the .GetAwaiter().GetResult().

Reply:
Changed to always return the Task, looks cleaner overall.